### PR TITLE
Fix coqdep on import categories and filters in Require

### LINKED ISSUE #16010: [coqdep] cannot parse import category
Here is an example:
```sh
$ cat foo.v
Require Import Sumbool(sumbool_of_bool).
$ coqdep foo.v
*** Error: File "foo.v",characters 22-23: Syntax error
```

cc @SkySkimmer

### PR DESCRIPTION
**Pull request by SkySkimmer**
Fix #16010

### CODE REVIEW

File: tools/coqdep/lexer.mll [hunk 1]
```diff
@@ -187,6 +191,18 @@ and comment = parse
   | _
       { comment lexbuf }
 
+and skip_parenthesized = parse
+  | "(*"
+      { comment lexbuf; skip_parenthesized lexbuf }
+  | "("
+      { skip_parenthesized lexbuf; skip_parenthesized lexbuf }
```
Comment (gares) [inline tools/coqdep/lexer.mll:198]: Do you really need nesting?

Comment (SkySkimmer) [reply tools/coqdep/lexer.mll:198]: We have the `module(inductive(..))` syntax
and maybe more complicated categories in the future

**Review by Alizter — APPROVED** [review]

File: tools/coqdep/lexer.mll [hunk 2]
```diff
@@ -187,6 +191,18 @@ and comment = parse
...
+  | ")"
+      { () }
+  | eof
+      { raise Fin_fichier }
+  | _
+    { skip_parenthesized lexbuf }
```
Comment (Alizter) [inline tools/coqdep/lexer.mll:204]: indent

File: tools/coqdep/lexer.mll [hunk 3]
```diff
@@ -252,6 +270,8 @@ and coq_qual_id_tail module_name = parse
 and coq_qual_id_list module_names = parse
   | "(*"
       { comment lexbuf; coq_qual_id_list module_names lexbuf }
+  | "("
+    { skip_parenthesized lexbuf; coq_qual_id_list module_names lexbuf }
```
Comment (Alizter) [inline tools/coqdep/lexer.mll:274]: indent

File: tools/coqdep/lexer.mll [hunk 4]
```diff
@@ -202,6 +218,8 @@ and load_file = parse
 and require_file from = parse
   | "(*"
       { comment lexbuf; require_file from lexbuf }
+  | "("
+    { skip_parenthesized lexbuf; require_file from lexbuf }
```
Comment (Alizter) [inline tools/coqdep/lexer.mll:222]: indent

**Comment by SkySkimmer** [comment]
Let's forget about these indent differences, they really don't matter

**Comment by Alizter** [comment]
OK I will fix them myself after then.

**Comment by Alizter** [comment]
merging soon

**Comment by ppedrot** [comment]
@Alizter I am planning to branch soon (probably tomorrow), can you merge this one quickly? It will save a backport. Thanks.

**Comment by Alizter** [comment]
@coqbot merge now

**Pull request merged**